x/x_test.go

```diff
@@ -52,3 +52,45 @@ func TestDivideAndRule(t *testing.T) {
 
 	test(1755, 4, 439)
 }
+
+func TestValidateAddress(t *testing.T) {
+	t.Run("IPv4", func(t *testing.T) {
+		testData := []struct {
+			name    string
+			address string
+			isValid bool
+		}{
+			{"Valid without port", "190.0.0.1", false},
+			{"Valid with port", "192.5.32.1:333", true},
+			{"Invalid without port", "12.0.0", false},
+			// the following test returns true because 12.0.0 is considered as valid
+			// hostname
+			{"Invalid with port", "12.0.0:3333", true},
```

Comment:
I am not sure if this test should return true. Is this the intended behaviour?
A similar test on IPv6 returns false (see https://github.com/dgraph-io/dgraph/pull/3027/files#diff-12d9d50e4e8f838a59b4ad0e7ad05251R87)